### DISCUSSION TITLE
Invalid enum value definitions error

### DISCUSSION THREAD

**Discussion by harbar20** [prisma/prisma#7757] [Q&A] [answered]
Hi everyone! I'm trying to take advantage of the enums feature to enforce some values in one of my models. This is my schema thus far:

```
enum System {
	d,
	c,
	dm,
	s,
	dl
}

enum KillType {
	d,
	p,
	n
}

enum PingTime {
	sent,
	first
}

model League {
	name      String
  	guildId   String
  	channelId String @id
  	resultsChannelId String?
  	dlId String?
  	sheetId String? 
  	system System @default("d")
}

model Rules {
	channelId String @id
	recoil KillType @default("d")
	suicide KillType @default("d")
	abilityitem KillType @default("p")
	selfteam KillType @default("n")
	db KillType @default("p")
	spoiler Boolean @default(true)
	ping String @default("")
	forfeit KillType @default("n")
	format String @default("default")
	quirks Boolean @default(true)
	stopTalking Boolean @default(false)
	tb Boolean @default(true)
	combine Boolean @default(false)
	redirect String @default("")
	timeOfPing PingTime @default("first")
}
```

but I got an error saying that my enum value definitions are invalid. Why are they, and what can I do to make them valid?

Edit: also, what can I use to automatically format Prisma in VS Code?

**Comment by ryands17** [comment, marked as answer]
@harbar20 👋 

Remove the commas in `enum` values and change the `@default` as follows:

```prisma
enum System {
  d
  c
  dm
  s
  dl
}

enum KillType {
  d
  p
  n
}

enum PingTime {
  sent
  first
}

model League {
  name             String
  guildId          String
  channelId        String  @id
  resultsChannelId String?
  dlId             String?
  sheetId          String?
  system           System  @default(d)
}

model Rules {
  channelId   String   @id
  recoil      KillType @default(d)
  suicide     KillType @default(d)
  abilityitem KillType @default(p)
  selfteam    KillType @default(n)
  db          KillType @default(p)
  spoiler     Boolean  @default(true)
  ping        String   @default("")
  forfeit     KillType @default(n)
  format      String   @default("default")
  quirks      Boolean  @default(true)
  stopTalking Boolean  @default(false)
  tb          Boolean  @default(true)
  combine     Boolean  @default(false)
  redirect    String   @default("")
  timeOfPing  PingTime @default(first)
}
```

I would suggest going through the docs for a better idea about enums and the schema:
https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#enum
https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
